hazelcast/include/hazelcast/client/topic/ReliableMessageListener.h

```diff
@@ -60,8 +60,19 @@ namespace hazelcast {
              *
              * @Beta
              */
-            class HAZELCAST_API ReliableMessageListener : public MessageListener {
+            class HAZELCAST_API ReliableMessageListener {
             public:
+                virtual ~ReliableMessageListener() = default;
+
+                /**
+                * Invoked when a message is received for the added topic. Note that topic guarantees message ordering.
+                * Therefore there is only one thread invoking onMessage. The user should not keep the thread busy, but preferably
+                * should dispatch it via an Executor. This will increase the performance of the topic.
+                *
+                * @param message the message that is received for the added topic
+                */
+                virtual void onMessage(Message &&message) = 0;
```

Comment:
i think this needs to match the signature of MessageListener on?received. Furthermore, lets try to remove the virtual methods where possible.

Reply:
I didn't wish to convert ReliableMessageListener on this PR, I just copy pasted the virtual methods from the old MessageListener and removed the inheritance so that it wouldn't break. I thought I would do it in a separate PR because ReliableMessageListener needs extra parameters other than handler functions (isLossTolerant, etc.) and I think we should discuss about how the API should look for those.

Reply:
can you open an issue to correct this API?